### DISCUSSION TITLE
Question about the Shadows activities handler

### DISCUSSION THREAD

**Discussion by tm6907** [cmu-sei/GHOSTS#405] [Q&A] [answered]
In this function found here: https://github.com/cmu-sei/GHOSTS/blob/master/src/ghosts.shadows/src/handlers/activities_handler.py

"""
def main(query, docs, documents):
    # documents = []
    for file in os.listdir("docs"):
        if file not in docs:
            docs.append(file)
            if file.endswith(".pdf"):
                pdf_path = "./docs/" + file
                loader = PyPDFLoader(pdf_path)
                documents.extend(loader.load())
                print(f"Loaded pdf {pdf_path}")
            elif file.endswith('.docx') or file.endswith('.doc'): ..........
"""

What is this "docs" referring to? As its failing when called as no docs are passed with the query to the handler from the api.py so it fails saying no such file or directory ./docs. It seems to pass an empty list from api.py.

Can't find anything in the documentation to help either.

**Comment by sei-dupdyke** [comment, marked as answer]
This is part of a RAG pipeline — we didn't include the docs we used internally for testing. You should be able to create that folder and have it continue (just not with any of that capability).

If you wanted to build a model to use or be smarter about a specific set of assets (like pdfs or word docs), you'd put them in this folder.

In a future update, we'll fix this so at least the folder is there, sorry.

**Reply by tm6907** [reply]
Can Pandora be used to negate this requirement?

**Reply by sei-dupdyke** [reply]
No, the two are not related in any way.
